test/mcd/mcd-tests.js

```diff
@@ -930,6 +939,330 @@ const mcdClaimTest = async () => {
+                    vaultId,
+                    Float2BN(boostAmount),
+                    ilkData.isCrop ? CROPPER_ADDR : MCD_MANAGER_ADDR,
+                    joinAddr,
+                    exchangeOrder,
+                )).wait();
+                const feesAfter = await balanceOf(DAI_ADDR, feeReciever);
+
+                const recipeEvent = receipt.events.find((e) => e.topics[0] === hre.ethers.utils.id('RecipeEvent(address,string)')
+                    && e.topics[1] === hre.ethers.utils.defaultAbiCoder.encode(['address'], [boostComposite.address])
+                    && e.topics[2] === hre.ethers.utils.id('McdBoostComposite'));
+                expect(recipeEvent).to.not.be.eq(undefined);
+
+                const actionEvent = receipt.events.find((e) => e.topics[0] === hre.ethers.utils.id('ActionEvent(string,bytes)')
+                    && e.address === boostComposite.address);
+                expect(actionEvent).to.not.be.eq(undefined);
+                const eventParams = hre.ethers.utils.defaultAbiCoder.decode(
+                    ['(address proxy, uint256 boostAmount, uint256 supplyAmount)'],
+                    hre.ethers.utils.defaultAbiCoder.decode(
+                        ['bytes'],
+                        actionEvent.data,
+                    )[0],
+                )[0];
+
+                const ratioAfter = await getRatio(mcdView, vaultId);
+                const info2 = await getVaultInfo(mcdView, vaultId, ilkData.ilkBytes);
+                console.log(
+                    `Ratio after: ${ratioAfter.toFixed(2)}% (coll: ${info2.coll.toFixed(2)} ${
+                        tokenData.symbol
+                    }, debt: ${info2.debt.toFixed(2)} Dai)`,
+                );
+                const feeAmount = Float2BN(boostAmount).div(400);
+
+                info.coll = Float2BN(`${info.coll}`, tokenData.decimals);
+                info.debt = Float2BN(`${info.debt}`);
+                info2.coll = Float2BN(`${info2.coll}`, tokenData.decimals);
+                info2.debt = Float2BN(`${info2.debt}`);
+
+                expect(ratioAfter).to.be.lt(ratioBefore);
+                expect(info2.coll.sub(info.coll)).to.be.closeTo(
+                    eventParams.supplyAmount,
+                    eventParams.supplyAmount.div(Float2BN('1', 6)),
+                );
+                expect(info2.debt.sub(info.debt)).to.be.closeTo(
+                    Float2BN(boostAmount),
+                    Float2BN(boostAmount).div(Float2BN('1', 6)),
+                );
+                expect(await balanceOf(tokenData.address, boostComposite.address)).to.be.eq(0);
+                expect(await balanceOf(DAI_ADDR, boostComposite.address)).to.be.eq(0);
+                expect(await balanceOf(tokenData.address, proxy.address)).to.be.eq(0);
+                expect(await balanceOf(DAI_ADDR, proxy.address)).to.be.eq(0);
+                expect(feesAfter.sub(feesBefore)).to.be.eq(feeAmount);
+            });
+        }
+    });
+};
+
```

Comment:
would create a test where a recipe includes composite actions just to be on the safe side